### ISSUE TITLE
Does not handle metadata that is an evaling_lit

### ISSUE THREAD

**Issue by sogaiu** [sogaiu/tree-sitter-clojure#50]
Came across [this code](https://github.com/oscaro/clj-gcloud-storage/blob/18a02a4bc08e47243cd936a54f8a5739b6437460/src/clj_gcloud/storage.clj#L51) today: 

```clojure
(defn get-bucket
  [^Storage storage ^String bucket-name]
  (let [^#=(array-type Storage$BucketGetOption) no-options (into-array Storage$BucketGetOption [])]
    (.get storage bucket-name no-options)))
```

Specifically this bit was not so familiar:

```clojure
^#=(array-type Storage$BucketGetOption) no-options
```

When this sort of thing is handed to clojure at the command line, one gets:

```
user=> ^#=(keyword "a") []
[]
```

To make things a bit more apparent:

```
user=> (binding [*print-meta* true] (prn (read-string "^#=(keyword \"a\") []")))
^{:a true} []
```

So I think that means that the `#=(...)` portion is supposed to be understood as what the grammar refers to as [`evaling_lit`](https://github.com/sogaiu/tree-sitter-clojure/blob/421546c2547c74d1d9a0d8c296c412071d37e7ca/grammar.js#L443-L450) [1].

The upcoming grammar doesn't handle this properly:

```
$ echo "^#=(keyword \"a\") []" | tree-sitter.ahlinc parse -
(source [0, 0] - [1, 0]
  (vec_lit [0, 0] - [0, 19]
    meta: (meta_lit [0, 0] - [0, 16]
      value: (tagged_or_ctor_lit [0, 1] - [0, 16]
        tag: (sym_lit [0, 2] - [0, 3]
          name: (sym_name [0, 2] - [0, 3]))
        value: (list_lit [0, 3] - [0, 16]
          value: (sym_lit [0, 4] - [0, 11]
            name: (sym_name [0, 4] - [0, 11]))
          value: (str_lit [0, 12] - [0, 15]))))))
```

Here the piece in question is being recognized as `tagged_or_ctor_lit` rather than `evaling_lit`.

---

[1] I think use of this is "discouraged" and I don't come across it much -- `project.clj` is where I'd seen it the most.

In Clojure's source, `LispReader.java` has some relevant bits  [here](https://github.com/clojure/clojure/blob/1e835a9d9fb4498a5cf643df861565c07701b18b/src/jvm/clojure/lang/LispReader.java#L115) and [here](https://github.com/clojure/clojure/blob/1e835a9d9fb4498a5cf643df861565c07701b18b/src/jvm/clojure/lang/LispReader.java#L1293-L1334).

**Comment by sogaiu** [comment]
This looks like it's in the same neighborhood as #46.

Looking back at https://github.com/sogaiu/tree-sitter-clojure/issues/35#issuecomment-1445367208, there is this alternate fix idea:

https://github.com/sogaiu/tree-sitter-clojure/blob/421546c2547c74d1d9a0d8c296c412071d37e7ca/grammar.js#L335-L342

...

> This brings up the topic of whether that choice should be there restricting things...possibly it could be replaced with $._form.

**Comment by sogaiu** [comment]
So I tried the following:
```javascript
    meta_lit: $ =>
    seq(field('marker', "^"),
        repeat($._gap),
        field('value', $._form)),

    old_meta_lit: $ =>
    seq(field('marker', "#^"),
        repeat($._gap),
        field('value', $._form)),
```

With that, the command line invocation from before became:
```
$ echo "^#=(keyword \"a\") []" | tree-sitter.ahlinc parse -
(source [0, 0] - [1, 0]
  (vec_lit [0, 0] - [0, 19]
    meta: (meta_lit [0, 0] - [0, 16]
      value: (evaling_lit [0, 1] - [0, 16]
        value: (list_lit [0, 3] - [0, 16]
          value: (sym_lit [0, 4] - [0, 11]
            name: (sym_name [0, 4] - [0, 11]))
          value: (str_lit [0, 12] - [0, 15]))))))
```

The Clojars testing results didn't change so I guess that may be a good sign.

The potential fix has been pushed as [a commit](https://github.com/sogaiu/tree-sitter-clojure/commit/2590a7f2a50b6c8ca05907047b5bbce3d6f17c01) to the pre-0.0.12 branch.

**Comment by sogaiu** [comment]
Addressed in aaf31ad73f8e83cd84d559d9b3f089cc064ac940.
